internal/datanode/index_services.go

```diff
@@ -497,6 +508,139 @@ func (node *DataNode) QueryJobsV2(ctx context.Context, req *workerpb.QueryJobsV2
 	}
 }
 
+func (node *DataNode) queryIndexTask(ctx context.Context, req *workerpb.QueryJobsRequest) (*workerpb.QueryJobsV2Response, error) {
+	log := log.Ctx(ctx).With(
+		zap.String("clusterID", req.GetClusterID()), zap.Int64s("taskIDs", req.GetTaskIDs()),
+	).WithRateGroup("QueryResult", 1, 60)
+
+	infos := make(map[typeutil.UniqueID]*index.IndexTaskInfo)
+	node.taskManager.ForeachIndexTaskInfo(func(ClusterID string, buildID typeutil.UniqueID, info *index.IndexTaskInfo) {
+		if ClusterID == req.GetClusterID() {
+			infos[buildID] = &index.IndexTaskInfo{
```

Comment:
use proto.Clone

Reply:
info is not workerpb.IndexTaskInfo, wrap a ToIndexTaskInfo func instead.